### PR TITLE
Pure Go cross-compile

### LINKED ISSUE #960: filebeat dynamically linked
When I download Filebeat 1.1.0 LINUX 64-BIT and do ```file filebeat```:
```
ELF 64-bit LSB  executable, x86-64, version 1 (SYSV), dynamically linked (uses shared libs), not stripped
```
What are the shared libs it's depending on?
Is it possible to be statically linked?

### PR DESCRIPTION
**Pull request by tsg**
Depends on #1666.

This produces statically linked binaries for Filebeat and Winlogbeat (which don't use CGo) by using simple cross compilation, without Cgo enabled. The advantage of the statically linked binaries is that they tend to be more portable.

Closes #960.

This also gets rid of the code duplication between `xgo-image` and `xgo-image-deb6`, at least for the `build.sh` script. It does that by simply copying the one from `xgo-image` over to `xgo-image-deb6`, I didn't have a better idea.

### CODE REVIEW

**Comment by tsg** [comment]
@cyrilleverrier you might be interested in this one.

**Comment by cyrilleverrier** [comment]
@tsg I had a quick look. LGTM.

File: dev-tools/packer/Makefile [hunk 1]
```diff
@@ -8,8 +8,8 @@ all: packetbeat/deb packetbeat/rpm packetbeat/darwin packetbeat/win packetbeat/b
 	winlogbeat/win \
 	build/upload/build_id.txt
 
-.PHONY: packetbeat topbeat filebeat metricbeat winlogbeat
-packetbeat topbeat filebeat metricbeat winlogbeat: build/upload
+.PHONY: packetbeat topbeat metricbeat winlogbeat
```
Comment (ruflin) [inline dev-tools/packer/Makefile]: winlogbeat should probably be removed here too

**Comment by ruflin** [comment]
LGTM besides one small commented.

File: dev-tools/packer/Makefile (hunk 1 above)
Comment (tsg) [inline dev-tools/packer/Makefile]: thanks, fixed.

**Comment by andrewkroh** [comment]
Fixes #960   👍 

LGTM

**Comment by andrewkroh** [comment]
jenkins, package it

**Comment by tsg** [comment]
jenkins, package it

**Pull request merged**